packages/theme-language-server-common/src/rename/providers/LiquidVariableRenameProvider.ts

```diff
@@ -0,0 +1,149 @@
+import { BaseRenameProvider } from '../BaseRenameProvider';
+import { DocumentManager } from '../../documents';
+import {
+  LiquidHtmlNode,
+  LiquidTagFor,
+  LiquidTagTablerow,
+  NamedTags,
+  NodeTypes,
+  Position,
+} from '@shopify/liquid-html-parser';
+import { Range } from 'vscode-languageserver';
+import {
+  PrepareRenameParams,
+  PrepareRenameResult,
+  RenameParams,
+  TextDocumentEdit,
+  TextEdit,
+  WorkspaceEdit,
+} from 'vscode-languageserver-protocol';
+import { visit } from '../../visitor';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { JSONNode } from '@shopify/theme-check-common';
+
+export class LiquidVariableRenameProvider implements BaseRenameProvider {
+  constructor(private documentManager: DocumentManager) {}
+
+  async prepare(
+    node: LiquidHtmlNode,
+    ancestors: LiquidHtmlNode[],
+    params: PrepareRenameParams,
+  ): Promise<null | PrepareRenameResult> {
+    const document = this.documentManager.get(params.textDocument.uri);
+    const textDocument = document?.textDocument;
+
+    if (!textDocument || !node || !ancestors) return null;
+    if (!supportedTags(node)) return null;
+
+    return {
+      range: Range.create(
+        textDocument.positionAt(node.position.start),
+        textDocument.positionAt(node.position.end),
```

Comment:
I feel like this is too far for an assignMarkup. Wouldn't that include the filters too? Wouldn't that mean you could rename by right clicking over contents after the `=`?